astronomer/providers/apache/livy/triggers/livy.py

```diff
@@ -0,0 +1,133 @@
+"""This module contains the Apache Livy Trigger."""
+import asyncio
+from typing import Any, AsyncIterator, Dict, Optional, Tuple, Union
+
+from airflow.triggers.base import BaseTrigger, TriggerEvent
+
+from astronomer.providers.apache.livy.hooks.livy import BatchState, LivyHookAsync
+
+
+class LivyTrigger(BaseTrigger):
+    """
+    Check for the state of a previously submitted job with batch_id
+
+    :param batch_id: Batch job id
+    :param spark_params: Spark parameters
+    :param livy_conn_id: reference to a pre-defined Livy Connection.
+    :param polling_interval: time in seconds between polling for job completion. Don't poll for values >=0
+    :param extra_options: A dictionary of options, where key is string and value
+        depends on the option that's being modified.
+    :param extra_headers: A dictionary of headers passed to the HTTP request to livy.
+    :param livy_hook_async: LivyHookAsync object
+    """
+
+    def __init__(
+        self,
+        batch_id: Union[int, str],
+        spark_params: Dict[Any, Any],
+        livy_conn_id: str = "livy_default",
+        polling_interval: int = 0,
+        extra_options: Optional[Dict[str, Any]] = None,
+        extra_headers: Optional[Dict[str, Any]] = None,
+        livy_hook_async: Optional[LivyHookAsync] = None,
+    ):
+        super().__init__()
+        self._batch_id = batch_id
+        self.spark_params = spark_params
+        self._livy_conn_id = livy_conn_id
+        self._polling_interval = polling_interval
+        self._extra_options = extra_options
+        self._extra_headers = extra_headers
+        self._livy_hook_async = livy_hook_async
+
+    def serialize(self) -> Tuple[str, Dict[str, Any]]:
+        """Serializes LivyTrigger arguments and classpath."""
+        return (
+            "astronomer.providers.apache.livy.triggers.livy.LivyTrigger",
+            {
+                "batch_id": self._batch_id,
+                "spark_params": self.spark_params,
+                "livy_conn_id": self._livy_conn_id,
+                "polling_interval": self._polling_interval,
+                "extra_options": self._extra_options,
+                "extra_headers": self._extra_headers,
+                "livy_hook_async": self._livy_hook_async,
+            },
+        )
+
+    async def run(self) -> AsyncIterator["TriggerEvent"]:  # type: ignore[override]
+        """
+        Checks if the _polling_interval > 0, in that case it pools Livy for
+        batch termination asynchrnonously.
+        else returns the success response
+        """
+        try:
+            if self._polling_interval > 0:
+                response = await self.poll_for_termination(self._batch_id)
+                if response.get("status") == "success":
```

Comment:
Doing the same thing in both if and else block?

Reply:
[Done](https://github.com/astronomer/astronomer-providers/pull/178/commits/f1f7fe1b5ab49260dfce31910577b5d35db4fbd0)